### PR TITLE
via support for bat43

### PR DESCRIPTION
**Pull request by yfuku**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/bat43/keymaps/via/keymap.c [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+enum layer_names {
+    _BASE,
+    _LOWER,
+    _RAISE
+};
+
+#define L_SPC LT(_LOWER, KC_SPC)
+#define R_ENT LT(_RAISE, KC_ENT)
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
```
Comment (drashna) [inline keyboards/bat43/keymaps/via/keymap.c:28]: This needs to have 4 layers (the default for dynamic keymaps), or you need to set the define in the config.h file to have only 3.

**Comment by yfuku** [comment]
@drashna Thanks for the review.
I'm sorry, I forgot.
I've added one more layer.

**Review by drashna — APPROVED** [review]
No worries. It's something that can be easily overlooked!

**Review by zvecr — APPROVED** [review]

**Comment by yfuku** [comment]
Thanks!

**Pull request merged**